File: spring-boot-postgresql/src/main/java/com/akvone/dev/DevStartupInitializer.java
package com.akvone.dev;

import com.akvone.data_jpa.entities.CityEntity;
import com.akvone.data_jpa.entities.StreetEntity;
import com.akvone.data_jpa.repositories.CityRepository;
import java.util.Collections;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.event.EventListener;

@Configuration
@RequiredArgsConstructor
public class DevStartupInitializer {

  private final CityRepository cityRepository;

  @EventListener(ApplicationReadyEvent.class)
  public void onApplicationReadyEvent() {
    CityEntity cityEntity = new CityEntity();
    cityEntity.setName("Moscow");
    StreetEntity streetEntity = new StreetEntity();
    streetEntity.setName("Lenina");
    cityEntity.setStreets(Collections.singletonList(streetEntity));

    cityRepository.save(cityEntity);
  }
}

File: spring-boot-postgresql/src/main/kotlin/com/akvone/dev/DevStartupInitializer.kt
package com.akvone.dev

import com.akvone.data_jpa.entities.CityEntity
import com.akvone.data_jpa.entities.StreetEntity
import com.akvone.data_jpa.repositories.CityRepository
import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.annotation.Configuration
import org.springframework.context.event.EventListener

@Configuration
class DevStartupInitializer(
    private val cityRepository: CityRepository
) {

    @EventListener(ApplicationReadyEvent::class)
    fun onApplicationReadyEvent() {
        val cityEntity = CityEntity()
        cityEntity.name = "Moscow"
        val streetEntity = StreetEntity()
        streetEntity.name = "Lenina"
        cityEntity.streets = listOf(streetEntity)
        cityRepository.save(cityEntity)
    }
}